symbolic-base/src/ZkFold/Symbolic/Data/UInt.hs

```diff
@@ -518,6 +520,30 @@ instance (Symbolic c, KnownNat n, KnownRegisterSize r) => StrictNum (UInt n r c)
                 return (p : ps <> [p'])
 
 
+instance
+  ( Symbolic c
+  , KnownNat n
+  , KnownNat (NumberOfRegisters (BaseField c) n r)
+  ) => SymbolicInput (UInt n r c) where
+  isValid (UInt bits) = Bool $ fromCircuitF bits solve
+    where
+        solve :: MonadCircuit i (BaseField c) m => Vector (NumberOfRegisters (BaseField c) n r) i -> m (Par1 i)
+        solve v = do
+            let len = Haskell.min (getNatural @n) (numberOfBits @(BaseField c))
+                vs = V.fromVector v
+            ys <- for vs $ \i -> do
+                                    bs <- bitsOf len i
```

Comment:
I think you meant `len` here to mean "number of bits in this register", which is not `min n numberOfBits` but `registerSize` and `highRegisterSize`